### ISSUE TITLE
rename column and self join issue

### ISSUE THREAD

**Issue by ericemc3** [uwdata/arquero#140]
Me again, i found an Observable notebook of mine broken today. I fixed it by adding a `reify()`.

This is about weddings by month, data includes total per year (MONTH="YR") that i need to put into a new column to compute a percentage:

```
dt = aq.table({"YEAR":[2017,2017,2017,2017,2017,2017,2017,2017,2017,2017,2017,2017,2017,2018,2018,2018,2018,2018,2018,2018,2018,2018,2018,2018,2018,2018],
"MONTH":["01","02","03","04","05","06","07","08","09","10","11","12","YR","01","02","03","04","05","06","07","08","09","10","11","12","YR"],
"WEDDINGS":["6074","7135","7990","17135","21101","29970","38346","27151","33071","14642","7857","10110","220582","5761","6764","9958","14973","22695","37403","29102","29159","32299","15116","8461","10462","222153"]})

dt
  .filter(d => d.MONTH == 'YR')
  .select('YEAR', {WEDDINGS: 'TOT_YEAR'})
  //.reify()       // **** i now need to add this one, otherwise TOT_YEAR appears undefined
  .join_right(dt.filter(d => d.MONTH != 'YR'))
```

**Comment by jheer** [comment]
Thanks! Alongside the performance updates for semijoin I did some refactoring that inadvertently caused a regression in the join algorithm. Sorry! Fixed in v4.1.4.

**Comment by ericemc3** [comment]
Thanks!
